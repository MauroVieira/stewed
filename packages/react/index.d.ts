/**
 * Required to avoid errors when importing these types of files
 */

declare module "*.svg?url" {
  const content: string;
  export default content;
}

declare module "*.svg" {
  const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
  export default content;
}

declare module "*.md" {
  const content: string;
  export default content;
}

declare module "*.scss" {
  const classes: { [key: string]: string };
  export default classes;
}
