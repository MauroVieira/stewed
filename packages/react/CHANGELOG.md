# @stewed/react

## 0.1.0

### Minor Changes

- a302a43: Update all packages

### Patch Changes

- Updated dependencies [a302a43]
  - @stewed/utilities@0.2.0
  - @stewed/hooks@0.1.0
  - @stewed/tokens@0.2.0
