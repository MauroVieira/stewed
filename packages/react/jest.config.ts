import { jestConfig } from "@harmonix-hub/testing";

module.exports = {
  ...jestConfig(),
  testEnvironment: "jsdom",
  moduleNameMapper: {
    "\\.(css|less|scss|sss|styl)$": "<rootDir>/node_modules/jest-css-modules",
  }
};
