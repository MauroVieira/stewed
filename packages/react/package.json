{
  "name": "@stewed/react",
  "version": "0.0.0",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "dist/**"
  ],
  "scripts": {
    "dev": "npx vite",
    "build": "npx vite build",
    "lint": "TIMING=1 eslint \"src/**/*.ts*\"",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@stewed/tsconfig": "workspace:*",
    "@stewed/utils": "workspace:*",
    "@stewed/lint": "workspace:*",
    "@types/react": "^18.0.9",
    "@types/react-dom": "^18.0.4",
    "eslint": "^8.15.0",
    "react": "^18.1.0",
    "tsup": "^5.10.1",
    "typescript": "^4.5.3",
    "sass": "^1.58.0",
    "@vitejs/plugin-react": "^1.3.2",
    "vite": "^2.9.15"
  },
  "publishConfig": {
    "access": "public"
  }
}
