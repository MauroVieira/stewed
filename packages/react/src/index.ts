export * from "./accordion";
export * from "./alert";
export * from "./aspect-ratio";
export * from "./avatar";
export * from "./badge";
export * from "./box";
export * from "./button";
export * from "./card";
export * from "./carousel";
export * from "./checkbox";
export * from "./divider";
export * from "./list-box";
export * from "./progress";
export * from "./radio";
export * from "./tabs";
export * from "./tag";
export * from "./text-area";
export * from "./text-field";
export * from "./theme";
export * from "./theme";
export * from "./toggle";
export * from "./typography";

// Utilities
export * from "./types";
