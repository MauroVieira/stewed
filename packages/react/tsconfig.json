{
  "extends": "@harmonix-hub/tsconfig/react.json",
  "compilerOptions": {
    "outDir": "./dist",
    "types": ["vitest/globals"]
  },
  "include": ["custom.d.ts", "index.ts", "./src/**/*", "./src/index.ts"],
  "exclude": ["dist", "src/**/stories", "src/**/__tests__", "node_modules"]
}
