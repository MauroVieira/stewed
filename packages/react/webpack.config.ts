import path from "path";
// Types
import type { Configuration } from "webpack";

const webpackConfig: Configuration = {
  mode: "production",
  entry: "./src/index.ts",
  output: {
    filename: "index.js",
    library: {
      type: "module",
    },
    path: path.resolve(__dirname, "dist"),
  },
  experiments: {
    outputModule: true,
  },
  optimization: {
    minimize: true,
  },
  module: {
    rules: [
      {
        test: /\.s[ac]ss$/i,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              modules: {
                auto: true,
                localIdentName: "[hash:base64]",
              },
              sourceMap: true,
            },
          },
          "sass-loader",
        ],
      },
      {
        test: /\.tsx?$/i,
        exclude: /node_modules/,
        use: {
          loader: "ts-loader",
          options: {
            compilerOptions: {
              sourceMap: true,
            },
          },
        },
      },
    ],
  },
  resolve: {
    extensions: [".tsx", ".ts", ".js"],
  }
};

export default webpackConfig;
