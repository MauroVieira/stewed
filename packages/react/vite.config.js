import { defineConfig } from "vite";
// Node
import { resolve } from "path";
// Plugins
import dts from "vite-plugin-dts";
import peerDepsExternal from "rollup-plugin-peer-deps-external";
import react from "@vitejs/plugin-react";
import { libInjectCss } from "vite-plugin-lib-inject-css";

export default defineConfig({
  plugins: [
    react(),
    libInjectCss(),
    dts({
      copyDtsFiles: true
    }),
  ],
  build: {
    copyPublicDir: false,
    lib: {
      entry: resolve(__dirname, "src/index.ts"),
      formats: ["es"],
    },
    rollupOptions: {
      plugins: [peerDepsExternal()],
      output: {
        assetFileNames: "assets/[name][extname]",
        entryFileNames: "[name].js",
      },
    },
  },
});
