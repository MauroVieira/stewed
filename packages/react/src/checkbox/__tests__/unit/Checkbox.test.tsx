import React from "react";
// UI Components
import { Checkbox, type CheckboxProps } from "../../index";
// Utilities
import { render, act, fireEvent } from "@testing-library/react";

describe("Checkbox", () => {
  describe("Snapshots", () => {
    it("should renders default component", () => {
      const { container } = render(<Checkbox>Checkbox</Checkbox>);

      expect(container).toMatchSnapshot();
    });

    it("should render additional classes", () => {
      const { container } = render(<Checkbox className="other-class">Checkbox</Checkbox>);

      expect(container).toMatchSnapshot();
    });

    it("should render loading", () => {
      const { container } = render(<Checkbox loading>Checkbox</Checkbox>);

      expect(container).toMatchSnapshot();
    });

    it.each<CheckboxProps["skin"]>(["primary", "critical", "success"])(
      "should apply '%s' skin classes",
      (skin) => {
        const { container } = render(<Checkbox skin={skin}>Checkbox</Checkbox>);

        expect(container).toMatchSnapshot();
      }
    );

    it.each<CheckboxProps["appearance"]>(["border", "default"])(
      "should apply '%s' appearance classes",
      (appearance) => {
        const { container } = render(<Checkbox appearance={appearance}>Checkbox</Checkbox>);

        expect(container).toMatchSnapshot();
      }
    );

    it.each<CheckboxProps["size"]>(["sm", "md", "lg"])("should apply '%s' size classes", (size) => {
      const { container } = render(<Checkbox size={size}>Checkbox</Checkbox>);

      expect(container).toMatchSnapshot();
    });
  });

  describe("Events", () => {
    it("should trigger `onChange` event when value changes", () => {
      const handleChange = vitest.fn();

      // Render the Checkbox component
      const { getByLabelText } = render(
        <Checkbox aria-label="checkbox-input" onChange={handleChange}>
          Checkbox
        </Checkbox>
      );

      // Get the Checkbox input by its ARIA label
      const input = getByLabelText("checkbox-input");

      // Simulate the change event
      act(() => {
        fireEvent.change(input, { target: { checked: true } });
      });

      // Check if the input is checked
      expect((input as HTMLInputElement).checked).toBe(true);
    });
  });
});
