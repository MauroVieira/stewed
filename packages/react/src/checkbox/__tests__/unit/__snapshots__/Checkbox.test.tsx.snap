// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > Snapshots > should apply 'border' appearance classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--primary_8d7086 _checkbox--border_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should apply 'critical' skin classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--critical_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should apply 'default' appearance classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--primary_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should apply 'lg' size classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--lg_8d7086 _checkbox--primary_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should apply 'md' size classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--primary_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should apply 'primary' skin classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--primary_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should apply 'sm' size classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--sm_8d7086 _checkbox--primary_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should apply 'success' skin classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--success_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should render additional classes 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--primary_8d7086 other-class"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should render loading 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--loading_8d7086 _checkbox--md_8d7086 _checkbox--primary_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <div
        class="_spinner_c80566 _spinner--white_c80566 _spinner--xs_c80566 _checkbox__spinner_8d7086"
      >
        <svg
          class="_spinner__svg_c80566"
          fill="currentColor"
          height="20"
          viewBox="0 0 48 48"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="24"
            cy="4"
            r="4"
          />
          <circle
            cx="12.19"
            cy="7.86"
            r="3.7"
          />
          <circle
            cx="5.02"
            cy="17.68"
            r="3.4"
          />
          <circle
            cx="5.02"
            cy="30.32"
            r="3.1"
          />
          <circle
            cx="12.19"
            cy="40.14"
            r="2.8"
          />
          <circle
            cx="24"
            cy="44"
            r="2.5"
          />
          <circle
            cx="35.81"
            cy="40.14"
            r="2.2"
          />
          <circle
            cx="42.98"
            cy="30.32"
            r="1.9"
          />
          <circle
            cx="42.98"
            cy="17.68"
            r="1.6"
          />
          <circle
            cx="35.81"
            cy="7.86"
            r="1.3"
          />
        </svg>
      </div>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;

exports[`Checkbox > Snapshots > should renders default component 1`] = `
<div>
  <label
    class="_checkbox_8d7086 _checkbox--md_8d7086 _checkbox--primary_8d7086"
  >
    <input
      class="_checkbox__input_8d7086"
      type="checkbox"
    />
    <span
      class="_checkbox__control_8d7086"
    >
      <svg
        class="_checkbox__svg_8d7086"
        fill="currentColor"
        height="20"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.723 6.237a.94.94 0 0 1 .053 1.277l-5.366 6.193a.834.834 0 0 1-.611.293.83.83 0 0 1-.622-.264l-2.927-3.097a.94.94 0 0 1 0-1.278.82.82 0 0 1 1.207 0l2.297 2.43 4.763-5.498a.821.821 0 0 1 1.206-.056Z"
        />
      </svg>
    </span>
    <div
      class="_checkbox__text_8d7086"
    >
      Checkbox
    </div>
  </label>
</div>
`;
